src/components/PrometheusGraph/_Chart.js

```diff
@@ -0,0 +1,209 @@
+import React from 'react';
+import styled from 'styled-components';
+import { debounce, sortedIndex, minBy } from 'lodash';
+import { line, area } from 'd3-shape';
+
+
+function getDatapointAtTimestamp(series, timestampSec) {
+  const timestamps = series.datapoints.map(d => d.timestampSec);
+  const index = sortedIndex(timestamps, timestampSec);
+  return series.datapoints[index];
+}
+
+const Canvas = styled.svg`
+  cursor: crosshair;
+  position: absolute;
+`;
+
+const SeriesLineChart = styled.path.attrs({
+  strokeWidth: 2,
+  fill: 'none',
+})`
+  opacity: ${props => (props.faded ? 0.1 : 1)};
+  pointer-events: none;
+`;
+
+const SeriesAreaChart = styled.path.attrs({
+  // Use strokeWidth only on focused area graphs to make sure ultra-thin ones
+  // still get visible, but don't use it when multiple series are visible so
+  // that it doesn't look like it's other series' border.
+  strokeWidth: ({ focused }) => (focused ? 1 : 0),
+  stroke: ({ fill }) => fill,
+})`
+  opacity: ${props => (props.faded ? 0.05 : 0.75)};
+  pointer-events: none;
+`;
+
+class Chart extends React.PureComponent {
+  constructor(props, context) {
+    super(props, context);
+
+    this.saveSvgRef = this.saveSvgRef.bind(this);
+    this.handleResize = debounce(this.handleResize.bind(this), 200);
+
+    this.handleGraphMouseMove = this.handleGraphMouseMove.bind(this);
+    this.handleGraphMouseLeave = this.handleGraphMouseLeave.bind(this);
```

Comment:
You can use the new autobind syntax: https://babeljs.io/docs/plugins/transform-class-properties/

Reply:
`transform-class-properties` plugin wasn't configured for this repo, but it was easy to add it so I addressed your comment :)

Reply:
Ah sorry I forgot which repo this was in!